server/src/main/java/org/opentosca/toscana/plugins/lifecycle/LifecycleAwarePlugin.java

```diff
@@ -0,0 +1,167 @@
+package org.opentosca.toscana.plugins.lifecycle;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.opentosca.toscana.core.plugin.AbstractPlugin;
+import org.opentosca.toscana.core.transformation.TransformationContext;
+import org.opentosca.toscana.core.transformation.platform.Platform;
+
+import org.slf4j.Logger;
+
+/**
+ This class represents the base class for every Plugin that wants to implement the Transformation lifecycle
+
+ @param <LifecycleT> The class type of the plugin specific implementation of the Lifecycle interface */
+public abstract class LifecycleAwarePlugin<LifecycleT extends TransformationLifecycle>
+    extends AbstractPlugin {
+
+    /**
+     Immutable list of execution tasks, they are in the right execution order and get executed from the first
+     index to the last
+     */
+    private List<ExecutionPhase<LifecycleT>> executionTasks;
+
+    /**
+     Initializes the plugin, that means that all tasks get added to a internal list (executionTasks) that then gets
+     executed by the transform method.
+     <p>
+     The tasks get stored in a list (environment tasks)
+     */
+    public LifecycleAwarePlugin(Platform platform) {
+        super(platform);
+        executionTasks = new ArrayList<>();
+
+        //Add the execution tasks
+        //Environment validation
+        executionTasks.add(new ExecutionPhase<>("check environment", (e) -> {
+            if (!checkEnvironment()) {
+                throw new ValidationFailureException("Transformation Failed," +
+                    " because the Environment check has failed!");
+            }
+        }));
+        //Node type Validation
+        executionTasks.add(new ExecutionPhase<>("check node types", this::checkNodeTypes));
+        //Model validation
+        executionTasks.add(new ExecutionPhase<>("check model", (e) -> {
+            if (!e.checkModel()) {
+                throw new ValidationFailureException("Transformation Failed," +
+                    " because the model check has failed!");
+            }
+        }));
+
+        addPostValidatePhases(executionTasks);
+
+        //Transformation phases
+        executionTasks.add(new ExecutionPhase<>("prepare", TransformationLifecycle::prepare));
+
+        addPostPreparePhases(executionTasks);
+
+        executionTasks.add(new ExecutionPhase<>("transformation", TransformationLifecycle::transform));
+
+        addPostTransformPhases(executionTasks);
+
+        executionTasks.add(new ExecutionPhase<>("cleanup", TransformationLifecycle::cleanup));
+
+        //Make list immutable
+        executionTasks = Collections.unmodifiableList(executionTasks);
```

Comment:
I'd use a local variable to build the list. `executionTasks` can then be made final.